src/components/Inventory/view/BatchesCellRenderer.tsx

```diff
@@ -41,29 +52,102 @@ export default function BatchesCellRenderer(props: RendererProps<TableRowType>):
     return <Typography className={classes.text}>{iValue}</Typography>;
   };
 
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    stopPropagation(event);
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const openChangeQuantityHandler = (type: string) => {
+    handleClose();
+    setModalValues({
+      openChangeQuantityModal: true,
+      type,
+    });
+  };
+
   if (column.key === 'withdraw') {
+    return (
+      <>
+        <CellRenderer
+          index={index}
+          column={column}
+          row={row}
+          value={
+            <Button
+              id='withdraw-batch'
+              label={strings.WITHDRAW}
+              onClick={rowClick}
+              size='small'
+              priority='secondary'
+              className={classes.text}
+            />
+          }
+        />
+      </>
+    );
+  }
+
+  if (column.key === 'batchNumber') {
+    return <CellRenderer {...props} value={createLinkToBatch(value)} />;
+  }
+
+  if (column.key === 'quantitiesMenu') {
     return (
       <CellRenderer
         index={index}
         column={column}
         row={row}
         value={
-          <Button
-            id='withdraw-batch'
-            label={strings.WITHDRAW}
-            onClick={rowClick}
-            size='small'
-            priority='secondary'
-            className={classes.text}
-          />
+          <>
+            <ChangeQuantityModal
+              open={modalValues.openChangeQuantityModal}
+              onClose={() => setModalValues({ openChangeQuantityModal: false, type: 'germinating' })}
+              modalValues={modalValues}
+              row={row as Batch}
+              reload={onRowClick}
+            />
+            <Popover
+              open={open}
+              anchorEl={anchorEl}
+              onClose={handleClose}
+              anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'left',
+              }}
+            >
+              <MenuList sx={{ padding: theme.spacing(2, 0) }}>
```

Comment:
i think clicking on the menu item is still marking row as selected.. may need to stop propagation from here.

Reply:
I think that if you press on the menu icon it's not marking the row as selected, if you do it outside the menu (most of the times, because the icon is pretty small) it gets clicked

Reply:
i meant the 2 menu items inside the menu

Reply:
Got it